app/controllers/api/v1/moves_controller.rb

```diff
@@ -22,10 +22,21 @@ def show
       def create
         move = Move.new(move_attributes)
         authorize!(:create, move)
-        move.save!
-        move.documents.each { |doc| doc.update(move: move) }
-        Notifier.prepare_notifications(topic: move, action_name: 'create')
-        render_move(move, 201)
+        if move.save
+          move.documents.each { |doc| doc.update(move: move) }
+          Notifier.prepare_notifications(topic: move, action_name: 'create')
+          render_move(move, 201)
+        elsif move.errors.details[:date].map { |x| x[:error] }.include?(:taken)
+          errors = validation_errors(move.errors)
+          errors.select { |e| e[:code] == :taken }.each do |error|
+            error.merge!(meta: { move_id: move.existing.id })
```

Comment:
This logic should not be here in the controller, 
at first sight it could go in the move model win a meaningful name